init.lua

```diff
@@ -2469,18 +2481,22 @@ command.add(
   end,
 
   ["lsp:rename-symbol"] = function(doc)
-    local symbol = doc:get_text(doc:get_selection())
-    local line1, col1, line2 = doc:get_selection()
-    if #symbol > 0 and line1 == line2 then
-      core.command_view:enter("New Symbol Name", {
-        text = symbol,
-        submit = function(new_name)
-          lsp.request_symbol_rename(doc, line1, col1, new_name)
-        end
-      })
-    else
-      core.log("Please select a symbol on the document to rename.")
-    end
+      local symbol = doc:get_text(doc:get_selection())
+      local line1, col1, line2, col2 = doc:get_selection()
+      if #symbol == 0 then
+        line1, col1, line2, col2 = get_token_range(doc, line1, col1)
```

Comment:
It looks like we might need to either change how Lite XL's tokenizer merges tokens (mainly how whitespace is merged with the next token), or we need to make `get_token_range` ignore the whitespace if possible.

Otherwise we might send a request with the position of a space, thus getting back something like "no symbol here".

Reply:
I wrote a small workaround, that trim white spaces before and after (even if after is not needed), but some times there are parenthesis and `;` that are attached to the token.

Using the plugin `evergreen` it improve the situation.